sklearn/ensemble/_gb.py

```diff
@@ -139,6 +135,39 @@ def update(self, j, est):
 class BaseGradientBoosting(BaseEnsemble, metaclass=ABCMeta):
     """Abstract base class for Gradient Boosting."""
 
+    _parameter_constraints = {
+        "learning_rate": [Interval(Real, 0.0, None, closed="neither")],
```

Comment:
The majority of these parameter constraints are for parameters being passed **directly** to `DecisionTreeRegressor`, so we can simplify a bit 
```suggestion
    _parameter_constraints = {
        **DecisionTreeRegressor._parameter_constraints,
        "learning_rate": [Interval(Real, 0.0, None, closed="neither")],
```

Note that you'll have to `pop` the `splitter` parameter from this since we hard-code it to `"best"` (side note: is there a good use case for allowing the user to set this? Not for this PR though.)

Reply:
I think the difference, at least for `min_samples_leaf`, is that GB implies in documentation that `min_samples_leaf=1.0` is a valid value, however even on `main` since `min_samples_leaf` is directly passed to `DecisionTreeRegressor` setting `min_samples_leaf=1.0` fails (because `DecisionTreeRegressor` validates it separately anyways).

For the direct-pass parameters I'm inclined to defer to the `DecisionTreeRegressor` unless there's specific computation or alteration within the GB estimator.

Reply:
Sounds reasonable. So should I update the docstrings as well? The other difference between GB and DTR is max_depth=None which isn't specified as a valid value in GB doscstrings.